src/test/pmem_memset/pmem_memset.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2014, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of Intel Corporation nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * pmem_memset.c -- unit test for doing a memset
+ *
+ * usage: pmem_memset <dest addr, int c, num bytes>
+ *
+ */
+
+#include "unittest.h"
+#include "libpmem.h"
+#include <stdio.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+int
+main(int argc, char *argv[])
+{
+	int fd;
+	struct stat stbuf;
+	void *dest;
+
+	START(argc, argv, "pmem_memset");
+
+	fprintf(stderr, "argc: %d\n", argc);
```

Comment:
Yes, I did. Thanks.

From: Lukasz Dorau [mailto:notifications@github.com]
Sent: Friday, January 9, 2015 2:52 AM
To: pmem/nvml
Cc: Jelinek, Sarah
Subject: Re: [nvml] code: Add support for pmem non-temporal memcpy, memmove and memset (#98)


In src/test/pmem_memset/pmem_memset.c<https://github.com/pmem/nvml/pull/98#discussion-diff-22707972>:

> +#include "libpmem.h"

> +#include <stdio.h>

> +#include <string.h>

> +#include <sys/stat.h>

> +#include <fcntl.h>

> +

> +int

> +main(int argc, char *argv[])

> +{

> +     int fd;

> +     struct stat stbuf;

> +     void *dest;

> +

> +     START(argc, argv, "pmem_memset");

> +

> +     fprintf(stderr, "argc: %d\n", argc);

I suppose you have forgotten to remove this debug trace?

—
Reply to this email directly or view it on GitHub<https://github.com/pmem/nvml/pull/98/files#r22707972>.